src/metabase/models/serialization.clj

```diff
@@ -1482,6 +1506,45 @@
   (set/union (viz-click-behavior-descendants  viz)
              (viz-column-settings-descendants viz)))
 
+;;; Common transformers
+
+(defn fk "Export Foreign Key" [model & [field-name]]
+  (cond
+    ;; this `::fk` is used in tests to determine that foreign keys are handled
+    (= model :model/User)  ^::fk [*export-user* *import-user*]
+    (= model :model/Table) ^::fk [*export-table-fk* *import-table-fk*]
+    (= model :model/Field) ^::fk [*export-field-fk* *import-field-fk*]
+    field-name             ^::fk [#(*export-fk-keyed* % model field-name) #(*import-fk-keyed* % model field-name)]
+    :else                  ^::fk [#(*export-fk* % model) #(*import-fk* % model)]))
+
+(defn nested "Nested entities" [model backward-fk opts]
+  (let [model-name (name model)
+        sorter     (:sort-by opts :created_at)
+        key-field  (:key-field opts :entity_id)]
+    ^::post
+    [(fn [data]
+       ;; if this looks weird, see :series hydration for DashboardCard, it supplies Cards while we need to serialize
+       ;; DashboardCardSeries instances
+       (let [entities (if (t2/instance-of? model (first data))
+                        data
+                        (t2/select model backward-fk (:id *current*)))]
```

Comment:
This branching, and then the usage of `t2/select` smells a bit to me. I wonder if there's a way to avoid it.

Firstly this means we'll have to do a call to `t2/select` for every card with a non-empty `:series`, where before we were using the hydrated dashboardcardseries that were hydrated on the dashboard here:
https://github.com/metabase/metabase/blob/150e97af24d6e0301f5025fb682723d777c9afb9/src/metabase/models/dashboard.clj#L588-L590
Using `t2/hydrate` meant the hydration of dashboardseries on dashcards was batched, so there would be one query per dashboard. Secondly it adds branching and makes use of dynamic scope, which complicate things a little. 

I think there could be a way around this if we avoid the use of `t2/hydrate` and instead implemented some simpler  methods for querying nested entities. The queries would preserve the data necessary for serialization.

Then you could change Dashboard's `serdes/extract-query` like so:
```
(defmethod serdes/extract-query "Dashboard" [_ opts]
  (eduction (map #(serdes/extract-nested %))
            (serdes/extract-query-collections Dashboard opts)))
```

You'd have to define `serdes/extract-nested` in terms of what entities are nested for the entity. Maybe you could declare the :nested entities somewhere like this:
```clojure
(defmethod serdes/make-spec "Dashboard" [_model-name opts]
  {:copy      [:archived :archived_directly :auto_apply_filters :cache_ttl :caveats :collection_position :created_at
               :description :embedding_params :enable_embedding :entity_id :initially_published_at :name
               :points_of_interest :position :public_uuid :show_in_getting_started :width]
   :skip      [;; those stats are inherently local state
               :view_count :last_viewed_at]
   :transform {:collection_id     (serdes/fk :model/Collection)
               :creator_id        (serdes/fk :model/User)
               :made_public_by_id (serdes/fk :model/User)
               :parameters        [serdes/export-parameters serdes/import-parameters]}
   :nested    {:dashcards {:model :model/DashboardCard,
                           :fk    :dashboard_id}
               :tabs      {:model :model/DashboardTab
                           :fk    :dashboard_id}}})
```
And have `serdes/extract-nested` use the `:nested` value above to hydrate `:dashcards` and `:tabs` or something.

I haven't thought through everything yet but wonder if you think this approach has legs.

Reply:
That's a good point. I'm thinking making all transforms a dictionary so it'll be easier to add more information that just a pair of functions, and then acting on that.

Also it seems to me that default `extract-query` could determine if there is a need to run `extract-nested` (or some kind of hydration or whatever) by itself, since `make-spec` doesn't really need a instance by itself.

Reply:
> extract-query could determine if there is a need to run extract-nested

It might also make sense to call map `extract-nested` over the results of `extract-query`, like this:

```clojure
(defmethod extract-all :default [model opts]
  (eduction (map (comp (partial log-and-extract-one model opts) (partial extract-nested model))
            (extract-query model opts)))
```

Reply:
> I'm thinking making all transforms a dictionary so it'll be easier to add more information

Yeah that could also work

Reply:
I've opted to overriding `:export` function on the DashboardCard spec, and `extract-nested` is going to be the next step (to remove dependency on hydration - which is indended for developers and not for serialization).